llvm/include/llvm/Support/GenericDomTreeConstruction.h

```diff
@@ -289,11 +289,6 @@ struct SemiNCAInfo {
         assert(NodeToInfo.contains(N) &&
                "ReverseChildren should not contain unreachable predecessors");
 
-        const TreeNodePtr TN = DT.getNode(N);
-        // Skip predecessors whose level is above the subtree we are processing.
-        if (TN && TN->getLevel() < MinLevel)
-          continue;
```

Comment:
Do we have some data to confirm this branch is dynamically dead? IIUC, this patch relies on tis being the case.

Reply:
I checked that if I replace this with an assertion, it does not fire in our test suite.

I opted not to keep the assertion, because this completely removes the dependency of runSemiNCA() on the dominator tree, which is imho cleaner.

Reply:
I think @alinas might have access to some fancy fleet-wide profiles -- could you confirm if this appears dead?